Sources/Clock.swift

```diff
@@ -25,15 +25,38 @@ public typealias AnnotatedTime = (
 /// print(Clock.now)
 /// ```
 public struct Clock {
+    private static var unfairLock = UnfairLock()
+    private static var _stableTime: TimeFreeze?
+    private static var _storage = TimeStorage(storagePolicy: .standard)
+
     private static var stableTime: TimeFreeze? {
-        didSet {
-            self.storage.stableTime = self.stableTime
+        get {
+            self.unfairLock.synchronized {
+                return self._stableTime
+            }
+        }
+        set {
+            self.unfairLock.synchronized {
+                self._stableTime = newValue
+                self._storage.stableTime = newValue
```

Comment:
[question] Why `_stableTime` is needed, when `_storage.stableTime` exists? I understand it was like this before, but I am struggling to understand why

Reply:
Storage effectively represents UserDefaults. I believe the goal is to avoid excessive user defaults reads, so you can think about `_stableTime` as in memory cache.